modules/actions/split.js

```diff
@@ -428,6 +428,9 @@ export function actionSplit(nodeIds, newWayIds) {
                             }
                         }
                     }
+                    if (way.tags.junction === 'roundabout' && way.isClosed()) {
```

Comment:
If the roundabout is a member of a `type=restriction`, `type=enforcement`, `type=junction` `junction=roundabout`, or `type=destination_sign` relation, but not a `type=route` relation, then it should be perfectly safe to split the roundabout. I’m pretty sure the code that handles restriction-like relation roles can handle closed ways just fine, but it would be a good case to test.

Reply:
Good point. `type=junction` should work as is.

I have to test this again, but I thought that the turn restriction editor would not properly handle this right now… I mean… does it even make sense to use a closed loop as a from/to/via member of a turn restriction or destination sign? :thinking:

Reply:
> does it even make sense to use a closed loop as a from/to/via member of a turn restriction or destination sign?

I don’t know how common it is, but picture a turning loop mapped as a closed way. Some turning loops allow you to go all the way around, while others do not. If not, the loop could be a member of a `no_u_turn` relation as both the `from` and `to`.

I’ve encountered this situation plenty of times at schools, but these tend to be parking lot roads that would at most be the beginning or end of a school bus route relation. Perhaps it would be more likely at a bus station?

A roundabout could be a likely `via` member, come to think of it. For example, if there’s a slip lane for going through, there might be a restriction prohibiting the user from using the roundabout itself to go through. I would always split a roundabout in such a scenario, but some mappers might prefer to keep the roundabout as a single closed way, for the same reason they’d do so in a route relation.

Reply:
I'm not sure I can quite follow your turning loop example. Do you have an example at hand how this looks like?

The roundabout-as-via situation is what I had in mind of not being properly handled currently: it's more or less the same issue as with route relations: the splitting would need to be more clever than what iD currently does [^1], as after the split the from -> via [--> via]* --> to path should "remain" contiguous. As far as I know it was never documented that closed roundabouts are allowed to be used in that way for turn restrictions, but examples [do exist (rarely)](https://overpass-turbo.eu/s/1Icb) (e.g., a closed roundabout as `to`: [relation/2996734](https://www.openstreetmap.org/relation/2996734), roundabouts as `via`: [relation/5378197](https://www.openstreetmap.org/relation/5378197), [relation/13437147](https://www.openstreetmap.org/relation/13437147)). FWIW, not even JOSM handles splitting those gray edge cases correctly :shrug: , but for iD I'd rather gravitate towards the more conservative approach of disallowing potentially data-breaking operations.

[^1]: Closed ways are split at the selected split point and the first/last node of the closed way.

Reply:
> I'm not sure I can quite follow your turning loop example. Do you have an example at hand how this looks like?

I’m not sure how to search for this, even though I encounter it from time to time. It would look something like this, but obviously in a more built-up environment. There would be a traffic sign indicating a No U Turn or No Left Turn restriction as you finish looping around:

![Turning circle](https://github.com/openstreetmap/iD/assets/1231218/4568792f-cf33-4f36-9d04-1dbd7e00f8d3)

Currently, creating a `restriction=no_u_turn` relation at the highlighted point would automatically split the closed way in two, which is fine. But if the roadway happens to be part of a bus route relation (not necessarily at the beginning or end of it), I don’t know if that would still happen automatically or if it would conflict with what you’ve written.

That said, I just noticed that it’s an and, not an or, so no turning loop would be affected.

I agree with taking a conservative approach. It’s just that iD unavoidably has a preference against closed roundabouts due to the ambiguity, so I’m a bit reticent about the possibility that it would become more difficult to resolve that ambiguity by splitting the way. Thinking through this is helping me feel better about it though.